buildSrc/src/main/java/org/elasticsearch/gradle/internal/InternalTestArtifactPlugin.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetContainer;
+import org.gradle.api.tasks.bundling.Jar;
+
+/**
+ * Ideally, yhis plugin is intended to be temporary and in the long run we want to move
+ * forward to port our test fixtures to use the gradle test fixtures plugin.
+ * */
+public class InternalTestArtifactPlugin implements Plugin<Project> {
+    @Override
+    public void apply(Project project) {
+        Configuration testArtifactsConfiguration = project.getConfigurations().create("testArtifacts");
+        testArtifactsConfiguration.extendsFrom(project.getConfigurations().getByName("testRuntimeClasspath"));
```

Comment:
Would `testRuntimeElements` be more appropriate here? Is there really a difference?

Reply:
I think I'm ending up redoing this by providing a variant aware solution. because that's what we really want here and what is the only _correct_ solution here imo. 
We want the testApiElements at compile time and testRuntimeElements at runtime. 
Also just using testRuntimeElements would bring us some other issue down the road where runtimeOnly elements would end up on test compile classpaths.